### PR TITLE
[Java][Spring][Server] Issue #4680

### PR DESCRIPTION
**Pull request by MZinchenko**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [x] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming langauge.

### Description of the PR

Issue #4680 fixed in proper way

### CODE REVIEW

**Comment by wing328** [comment]
@MZinchenko thanks for the PR. Please run `./bin/spring-all-pestore.sh` to update Spring Petstore samples so that CI can verify the change.

**Comment by MZinchenko** [comment]
Sorry for long time reply.
I found mistake in my merge and fix it.

Also when I update java samples, my samples\server\petstore\spring-mvc\pom.xml always going wrong.
Artefact and module name changes to swagger-spring (must be swagger-spring-mvc-server).
Am I doing smth wrong?

**Comment by wing328** [comment]
> Artefact and module name changes to swagger-spring (must be swagger-spring-mvc-server).

I'll take a look...

**Comment by wing328** [comment]
I'm doing some tests as well...

cc @cbornet

**Comment by wing328** [comment]
Tests passed via https://circleci.com/gh/swagger-api/swagger-codegen/2594 (after updating Spring Petstore samples)

**Comment by cbornet** [comment]
The samples of this PR are not OK so it's hard to review...

**Comment by MZinchenko** [comment]
What's wrong in samples?

2017-11-15 18:13 GMT+03:00 Christophe Bornet <notifications@github.com>:

**Comment by cbornet** [comment]
There are @Generated annotations everywhere instead of being hidden, the examples have been removed from spring controllers, etc...

**Comment by MZinchenko** [comment]
I simply ran scripts:
java-petstore.bat
java-petstore-okhttp-gson.bat
java-play-framework-petstore-server.bat
java-vertx-async-petstore-server.bat
java-vertx-rx-petstore-server.bat
spring-mvc-petstore-j8-async-server.bat
spring-mvc-petstore-server.bat
springboot-petstore-server.bat

What can I do to solve this problems?
Annotation @Generated <https://github.com/generated> created by default, as
I can see (in my swagger classes it exists too)

2017-11-15 23:49 GMT+03:00 Christophe Bornet <notifications@github.com>:

**Comment by cbornet** [comment]
The windows scripts are incorrect...

**Comment by MZinchenko** [comment]
Ok. I fix windows scripts and push it in this PR.
But as I can see there are lot of changes is samples.
There are comment changes and "get => is" at most.
As I can see this is result of changing version from 2.2.3 to 2.3.0.


2017-11-17 1:56 GMT+03:00 Christophe Bornet <notifications@github.com>:

File: samples/server/petstore/spring-mvc-j8-async/src/main/java/io/swagger/api/PetApi.java [hunk 1]
```diff
@@ -61,7 +61,8 @@
         produces = { "application/xml", "application/json" }, 
         consumes = { "application/json", "application/xml" },
         method = RequestMethod.POST)
-    default CompletableFuture<ResponseEntity<Void>> addPet(@ApiParam(value = "Pet object that needs to be added to the store" ,required=true )  @Valid @RequestBody Pet body) {
+    default CompletableFuture<ResponseEntity<Void>> addPet(
+@ApiParam(value = "Pet object that needs to be added to the store" ,required=true )  @Valid @RequestBody Pet body) {
```
Comment (cbornet) [inline samples/server/petstore/spring-mvc-j8-async/src/main/java/io/swagger/api/PetApi.java]: Alignment is wrong

File: samples/server/petstore/spring-mvc-j8-async/src/main/java/io/swagger/api/PetApi.java [hunk 2]
```diff
@@ -81,7 +82,9 @@
     @RequestMapping(value = "/pet/{petId}",
         produces = { "application/xml", "application/json" }, 
         method = RequestMethod.DELETE)
-    default CompletableFuture<ResponseEntity<Void>> deletePet(@ApiParam(value = "Pet id to delete",required=true) @PathVariable("petId") Long petId,@ApiParam(value = "" ) @RequestHeader(value="api_key", required=false) String apiKey) {
+    default CompletableFuture<ResponseEntity<Void>> deletePet(@ApiParam(value = "Pet id to delete",required=true) @PathVariable("petId") Long petId
+,
```
Comment (cbornet) [inline samples/server/petstore/spring-mvc-j8-async/src/main/java/io/swagger/api/PetApi.java]: This comma should not be here

File: samples/server/petstore/spring-mvc-j8-async/src/main/java/io/swagger/api/PetApi.java [hunk 3]
```diff
@@ -102,7 +105,8 @@
     @RequestMapping(value = "/pet/findByStatus",
         produces = { "application/xml", "application/json" }, 
         method = RequestMethod.GET)
-    default CompletableFuture<ResponseEntity<List<Pet>>> findPetsByStatus(@NotNull @ApiParam(value = "Status values that need to be considered for filter", required = true, allowableValues = "available, pending, sold") @Valid @RequestParam(value = "status", required = true) List<String> status) {
+    default CompletableFuture<ResponseEntity<List<Pet>>> findPetsByStatus(@NotNull @ApiParam(value = "Status values that need to be considered for filter", required = true, allowableValues = "available, pending, sold") @Valid @RequestParam(value = "status", required = true) List<String> status
+) {
```
Comment (cbornet) [inline samples/server/petstore/spring-mvc-j8-async/src/main/java/io/swagger/api/PetApi.java]: This should not be on this line

File: samples/server/petstore/spring-mvc-j8-async/src/main/java/io/swagger/configuration/SwaggerDocumentationConfig.java [hunk 4]
```diff
@@ -17,7 +17,7 @@
     ApiInfo apiInfo() {
         return new ApiInfoBuilder()
             .title("Swagger Petstore")
-            .description("This spec is mainly for testing Petstore server and contains fake endpoints, models. Please do not use this for any other purpose. Special characters: \" \\")
+            .description("This is a sample server Petstore server.  You can find out more about Swagger at [http://swagger.io](http://swagger.io) or on [irc.freenode.net, #swagger](http://swagger.io/irc/).  For this sample, you can use the api key `special-key` to test the authorization filters.")
```
Comment (cbornet) [inline samples/server/petstore/spring-mvc-j8-async/src/main/java/io/swagger/configuration/SwaggerDocumentationConfig.java]: The swagger spec used seems different than before. Still an issue with Windows script ?

File: samples/server/petstore/spring-mvc/src/main/java/io/swagger/api/StoreApi.java [hunk 5]
```diff
@@ -52,10 +53,11 @@
         @ApiResponse(code = 200, message = "successful operation", response = Order.class),
         @ApiResponse(code = 400, message = "Invalid ID supplied"),
         @ApiResponse(code = 404, message = "Order not found") })
-    @RequestMapping(value = "/store/order/{order_id}",
+    @RequestMapping(value = "/store/order/{orderId}",
         produces = { "application/xml", "application/json" }, 
         method = RequestMethod.GET)
-    ResponseEntity<Order> getOrderById(@Min(1) @Max(5) @ApiParam(value = "ID of pet that needs to be fetched",required=true) @PathVariable("order_id") Long orderId);
+    ResponseEntity<Order> getOrderById(@Min(1) @Max(5) @ApiParam(value = "ID of pet that needs to be fetched",required=true) @PathVariable("orderId") Long orderId
+);
```
Comment (cbornet) [inline samples/server/petstore/spring-mvc/src/main/java/io/swagger/api/StoreApi.java]: Should not be on this line

**Comment by MZinchenko** [comment]
Summary, there are some problems:
1) spring-mvc-petstore-server.sh, springboot-petstore-server.sh and other
java shell scripts using
"petstore-with-fake-endpoints-models-for-testing.yaml" but "petstore.yaml".
That's very strange (windows scripts using "petstore.yaml").
2) windows scripts are very outdated. I think, they must be removed or
updated. I can't run shell scripts nativly, but I'll try
3) suddenly new-line problem. I can't understand why new-lines appears
before some params.
    I revert my branch changes, but they are still here. I think this
problem is elder or environment aware.

2017-11-17 19:15 GMT+03:00 Christophe Bornet <notifications@github.com>:

**Comment by cbornet** [comment]
Correct and most used scripts are Unix ones. I think the windows scripts have been added as a block without checking each particular languages so that's why there are differences.

File: modules/swagger-codegen/src/main/resources/JavaSpring/pathParams.mustache [hunk 6]
```diff
@@ -1 +1 @@
-{{#isPathParam}}{{#useBeanValidation}}{{>beanValidationPathParams}}{{/useBeanValidation}}@ApiParam(value = "{{{description}}}"{{#required}},required=true{{/required}}{{#allowableValues}}, allowableValues = "{{#values}}{{{.}}}{{^-last}}, {{/-last}}{{#-last}}{{/-last}}{{/values}}"{{/allowableValues}}{{#defaultValue}}, defaultValue="{{{defaultValue}}}"{{/defaultValue}}) @PathVariable("{{baseName}}") {{>optionalDataType}} {{paramName}}{{/isPathParam}}
\ No newline at end of file
+{{#isPathParam}}{{#useBeanValidation}}{{>beanValidationPathParams}}{{/useBeanValidation}}@ApiParam(value = "{{{description}}}"{{#required}},required=true{{/required}}{{#allowableValues}}, allowableValues = "{{#enumVars}}{{#lambdaEscapeDoubleQuote}}{{{value}}}{{/lambdaEscapeDoubleQuote}}{{^-last}}, {{/-last}}{{#-last}}{{/-last}}{{/enumVars}}"{{/allowableValues}}{{#defaultValue}}, defaultValue="{{{defaultValue}}}"{{/defaultValue}}) @PathVariable("{{baseName}}") {{>optionalDataType}} {{paramName}}{{/isPathParam}}
```
Comment (cbornet) [inline modules/swagger-codegen/src/main/resources/JavaSpring/pathParams.mustache]: This is were the extra line has been added. Remove it and regenerate the samples.

Comment (MZinchenko) [reply modules/swagger-codegen/src/main/resources/JavaSpring/pathParams.mustache]: You means new line after {{/isPathParam}}?
I already done that and regenerate the samples, but new lines not removed.
Also I revert all changes in my branch and regenerate the samples, but new lines not removed. 
As you can see, new lines appears in blocks generated by other templates, not pathParams.mustache only. First of all PetApi.java using bodyParams.mustache for addPet method, but I don't  change bodyParams.mustache

Comment (MZinchenko) [reply modules/swagger-codegen/src/main/resources/JavaSpring/pathParams.mustache]: I pushed this change and some windows scripts changes, but samples still have some differences

**Comment by MZinchenko** [comment]
Ok. I update samples under CentOS inside Docker.
New lines vanished, so that's some type of environment bug.
That's no good, some users may use codegen under Windows with default setting and they are suffering.

**Comment by wing328** [comment]
@MZinchenko PR merged into master. Thanks for your contribution.

When you've time, I wonder if you can contact me via the email address on my Github profile. Thanks.

**Pull request merged**